src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -31,39 +48,48 @@ public String todayDate(DateTimePart datePart) {
      *                   - 3-й элемент массива - день (число);
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        if (dateParams.length != 3) {
+            return Optional.empty();
+        }
+        Optional<LocalDate> result;
+        try {
+            result = Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
+        } catch (DateTimeException e) {
+            result = Optional.empty();
```

Comment:
don't build logic using try catch block, use two returns (in try block and after catch)

Reply:
done